src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -1,41 +1,73 @@
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.stereotype.Service;
 
+import java.time.Instant;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
 
+@Service
+@RequiredArgsConstructor
 public class BasicChannelService implements ChannelService {
-    private static volatile BasicChannelService instance;
+
     private final ChannelRepository channelRepository;
+    private final ReadStatusRepository readStatusRepository;
+    private final MessageRepository messageRepository;
+    private final ChannelMapper channelMapper;
+    private final MessageSource messageSource;
 
-    public BasicChannelService(ChannelRepository channelRepository) {
-        this.channelRepository = channelRepository;
-    }
 
-    protected static BasicChannelService getInstance(ChannelRepository channelRepository) {
-        if (instance == null) {
-            synchronized (BasicChannelService.class) {
-                if (instance == null) {
-                    instance = new BasicChannelService(channelRepository);
-                }
-            }
-        }
-        return instance;
+    @Override
+    public CreatePrivateChannelResponse createPrivateChannel(CreatePrivateChannelRequest request) {
+        Channel channel = Channel.builder().
+                type(ChannelType.PRIVATE)
+                .build();
+        Channel savedChannel = channelRepository.save(channel);
+
+        request.getChannelMembersIds().forEach(memberId ->
+                readStatusRepository.save(new ReadStatus(memberId, channel.getId()))
+        );
+
+        Set<UUID> usersByChannelId = readStatusRepository.findUsersByChannelId(savedChannel.getId());
+        return channelMapper.toPrivateResponse(savedChannel,usersByChannelId);
     }
 
     @Override
-    public Channel createChannel(String name, String topic, ChannelType type) {
-        return channelRepository.save(new Channel(name, topic, type));
+    public CreatePublicChannelResponse createPublicChannel(CreatePublicChannelRequest request) {
+        Channel channel = Channel.builder()
+                .name(request.getName())
+                .topic(request.getTopic())
+                .type(ChannelType.PUBLIC).build();
+        Channel savedChannel = channelRepository.save(channel);
+
+        return channelMapper.toPublicResponse(channel);
     }
 
-    @Override
-    public Optional<Channel> getChannelDetails(UUID id) {
-        return channelRepository.findById(id);
+    public FindPrivateChannelResponse findPrivateChannel(UUID privateChannelId){
+        Optional<Channel> optChannel = channelRepository.findById(privateChannelId);
+        if (optChannel.isEmpty()) {
+            throw new IllegalArgumentException(messageSource.getMessage("error.channel.id.notfound", new Object[]{privateChannelId}, LocaleContextHolder.getLocale()));
+        }
```

Comment:
```suggestion
        Channel channel = channelRepository.findById(privateChannelId)
            .orElseThrow(() -> new IllegalArgumentException(messageSource.getMessage("error.channel.id.notfound", new Object[]{privateChannelId}, LocaleContextHolder.getLocale())));
```
if문을 쓰지 않아도 됩니다! 아래 내용 참고해보세요ㅎㅎ

https://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/